test/test-global-compress.js

```diff
@@ -29,6 +29,7 @@ test('should send a deflated data', t => {
   }, (err, res) => {
     t.error(err)
     t.strictEqual(res.headers['content-encoding'], 'deflate')
+    t.strictEqual(res.headers.vary, 'accept-encoding')
```

Comment:
I have added this here, since I don't think it deserves its own test, since that test will mostly be duplicated code.